api/api.go

```diff
@@ -0,0 +1,45 @@
+package api
```

Comment:
Hmm, it feels a little odd to have the MS graph client in an "api" package while the other clients are in the top-level "azureauth" package (azure.go file). I know the package structure is like this in Azure secrets, but I think we should do what's best to help with readability/comprehension in the context of this repo.

Open to ideas! One idea would be a "client" package to organize related client code. Another idea would be to have all client-related code in the azure.go file.

Reply:
I do like the idea of having all client implementations in a separate package for readability and maintenance. `azure.go` is already quite populated with the implementation and setup of the Azure Provider, so it would make it pretty big and make it harder to find things in that file if we move everything in there 😅 

I like the idea of a `client` package which separates all the clients from the Azure Provider interface and its setup, and I'm also open to thoughts from other folks on the team! :)

Reply:
I think having a client package and moving everything there makes the most sense. But maybe that should be scoped to another task? We could just rename `api` -> `client` for now.

Reply:
+1 on @fairclothjm idea. We can rename the api package to client. Then we can create a task to move the rest of the client code into that package (after 1.13 when we have more time).

Reply:
Renamed the package to `client` and also renamed `api.go` -> `client.go`. Thanks for the feedback! 😄